catalog/resource_online_table.go

```diff
@@ -80,13 +80,14 @@ func ResourceOnlineTable() common.Resource {
 			if err != nil {
 				return err
 			}
+			// Note: We should set the id right after creation and before waiting for online table to be available.
+			// This is because in case when online table isn't availabe, we still should have that resource in the state.
```

Comment:
nit: slightly more clear/precise way of explaining this
```suggestion
			// If the resource creation timeout is exceeded while waiting for the online table to be ready, this ensures the online table is persisted in the state.
```